lib/propolis/src/block/file.rs

```diff
@@ -45,10 +44,19 @@ impl FileBackend {
         let p: &Path = path.as_ref();
 
         let meta = metadata(p)?;
-        let read_only = readonly || meta.permissions().readonly();
+        let read_only = match (opts.read_only, meta.permissions().readonly()) {
+            (Some(false), true) => Err(Error::new(
+                ErrorKind::Other,
+                "writeable backend with read-only file not allowed",
+            )),
+            (Some(ro), false) => Ok(ro),
+            (_, file_ro) => Ok(file_ro),
+        }?;
 
         let fp = OpenOptions::new().read(true).write(!read_only).open(p)?;
-        let len = fp.metadata().unwrap().len() as usize;
+        let len = fp.metadata().unwrap().len();
+        // TODO: attempt to query blocksize from underlying file/zvol
+        let block_size = opts.block_size.unwrap_or(512);
```

Comment:
Don't wanna use `DEFAULT_BLOCK_SIZE` here too?

Reply:
Waffled on it, so.... done.